cmd/lifecycle/analyzer.go

```diff
@@ -128,34 +166,40 @@ func (a *analyzeCmd) Exec() error {
 	return nil
 }
 
-func (aa analyzeArgs) analyze(group buildpack.Group, cacheStore lifecycle.Cache) (platform.AnalyzedMetadata, error) {
+func (aa analyzeArgs) analyze() (platform.AnalyzedMetadata, error) {
 	var (
 		img imgutil.Image
 		err error
 	)
-	if aa.useDaemon {
-		img, err = local.NewImage(
-			aa.imageName,
-			aa.docker,
-			local.FromBaseImage(aa.imageName),
-		)
-	} else {
-		img, err = remote.NewImage(
-			aa.imageName,
-			aa.keychain,
-			remote.FromBaseImage(aa.imageName),
-		)
-	}
-	if err != nil {
-		return platform.AnalyzedMetadata{}, cmd.FailErr(err, "get previous image")
+	if aa.imageName != "" {
```

Comment:
I believe this will only ever be true for platform API > 0.7. The `-previous-image` flag is optional and maps to to `imageName`. Previously platforms would require the argument `<image>` for analyzer.

Reply:
I think at this point in the code `aa.imageName` already holds the value of either `-previous-image` or `<image>`.

Reply:
I think there was a miss understanding here.
I think that `aa.imageName != ""` will always be true for `platform API < 0.7` and it might be true for `platform API >= 0.7`.
@jabrown85 @natalieparellano, do you agree?